Hazel/src/Platform/OpenGL/OpenGLVertexArray.cpp

```diff
@@ -71,8 +73,8 @@ namespace Hazel {
 
 	void OpenGLVertexArray::SetIndexBuffer(const std::shared_ptr<IndexBuffer>& indexBuffer)
 	{
-		glBindVertexArray(m_RendererID);
-		indexBuffer->Bind();
+		glVertexArrayElementBuffer(m_RendererID,
+			std::dynamic_pointer_cast<OpenGLIndexBuffer>(indexBuffer)->m_RendererID);
```

Comment:
Same reasoning, convert to a static_pointer_cast.

```suggestion
			std::static_pointer_cast<OpenGLIndexBuffer>(indexBuffer)->m_RendererID);
```